lua/neo-tree/sources/common/components.lua

```diff
@@ -15,9 +15,54 @@ local utils = require("neo-tree.utils")
 local file_nesting = require("neo-tree.sources.common.file-nesting")
 local container = require("neo-tree.sources.common.container")
 local log = require("neo-tree.log")
+local path = require("plenary.path")
 
 local M = {}
 
+-- Workaround for https://github.com/nvim-lua/plenary.nvim/issues/411
+local relative_path = function(original_path, reference_path)
+  -- Use plenary's make_relative to clean paths
+  original_path = path:new(original_path):make_relative(".")
+  reference_path = path:new(reference_path):make_relative(".")
+  local o_path = path:new(original_path)
+  local ref_path = path:new(reference_path)
+  local parents = o_path:parents()
+  local ref_parents = ref_path:parents()
+
+  local path_elements = vim.split(o_path.filename, "/")
```

Comment:
Windows uses backslashes, so this might not always work. to take that into account:

```suggestion
  local sep = o_path.sep
  local path_elements = vim.split(o_path.filename, sep)
```